pkg/render/common/components/components.go

```diff
@@ -158,9 +158,14 @@ func GetPriorityClassName(overrides any) string {
 	if !value.IsValid() {
 		return ""
 	}
-	return value.String()
+	if value.Kind() == reflect.String {
+		return value.Interface().(string)
+	}
+	return ""
 }
 
+var getFieldHookUT = func(fieldNames []string) {}
```

Comment:
I know this doesn't affect the actual functionality of the code, but this sort of global variable / function setting, even just for tests, can lead to unexpected behaviour. This function will still be set when a test that sets it ends, and it could block (in the case that channels were used) or panic depending on how it's defined and used outside of the intended scope.

Reply:
I suppose I could just hardcode the handled fields in components_test.go, instead of "discovering" them dynamically in this way.  WDYT?

Reply:
I suppose we could, but that sort leaves a different problem of the tests not aligning with code changes implicitly (which I guess would be worse?).